src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/infrastructure/client/FeignErrorDecoder.java

```diff
@@ -0,0 +1,39 @@
+package uk.gov.hmcts.reform.iahomeofficeintegrationapi.infrastructure.client;
+
+import feign.Response;
+import feign.codec.ErrorDecoder;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.server.ResponseStatusException;
+
+@Component
+@Slf4j
+public class FeignErrorDecoder implements ErrorDecoder {
+
+    @Override
+    public Exception decode(String methodKey, Response response) {
+
+        switch (response.status()) {
+            case 400:
+            case 404:
+                try {
+                    log.error("StatusCode: {}, methodKey: {}, reason: {}, message: {}",
+                        response.status(),
+                        methodKey,
+                        response.reason(),
+                        IOUtils.toString(response.body().asReader(Charset.defaultCharset())));
```

Comment:
You should be able to decode the message better if you look at the API there is a structure to it with a code, text etc.

Reply:
I will address this in PR #77

Reply:
Please link to a line in another PR where this is fixed.

Reply:
I have made the changes in this PR.